README.md

```diff
@@ -1,6 +1,394 @@
+  The response is a json object listing all Services and Plans offered by the Broker.
+  The Controller Manager processes this response into a set of *ServiceClasses* in the API Server.
+  In this case, our *ServiceClass* is a provisionable object bucket.
+  A *ServiceClass* can provision many *ServiceInstances* of its class.
+
+- *ServiceInstance*
+
+  Service Catalog representation of a consumable service instance in the External Service Provider.
+  In this case, a single object bucket.
+  A *ServiceInstance* can have many *ServiceInstanceCredentials*, so long as the service supports this.
+  This enables a single instance to be consumed by many Pods.
+
+- *ServiceInstanceCredential*
+
+  Service Catalog object that does not contain any authentication or coordinate information.
+  Instead, when a *ServiceInstanceCredential* is created in the Service Catalog API Server, it triggers a request to the Broker for authentication and coordinate information.
+  Once a response is received, the sensitive information is stored in a *Secret* in the same namespace as the *ServiceInstanceCredential*.
+
+### What It Does
+This broker implements [OpenServiceBroker](https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md) methods for creating, connecting to, and destroying
+object buckets.
+- For each new *ServiceInstance*, a new, uniquely named bucket is created.
+- For each new *ServiceInstanceCredential*, a Secret is generated with the
+coordinates and credentials of the *ServiceInstance's* associated bucket.
+- Deleting a *ServiceInstance* destroys the associated bucket.
+- Deleting a *ServiceInstanceCredential* deletes the secret.
+
+### Limitations
+
+- Currently, the CNS Object Broker is dependent on GCE.  If run outside of a GCE environment, it will fail to start because the CNS Object Broker detects the external IP of the node on which it is run by calling to the GCP metadata server.
+It pairs this IP with the port of the `gluster-s3-deployment` *Service* to generate the coordinaates returned in the *ServiceInstanceCredential*.  This is not a requirement of brokers in general.
+
+- As it is implemented, the CNS Object Broker must run on the same Kubernetes cluster as the `gluster-s3-deployment` because
+it accesses the *Service* that exposes its *NodePort*.  This is done through a kubernetes api client, which will fail if it cannot
+ reach the `gluster-s3-deployment`'s *Service*. This is not a requirement of brokers in general.
+
+- Auth:  The S3 API implementation (Gluster-Swift) does not enforce any authentication / authorization.
+Each new bucket, regardless of the *Namespace* of its *ServiceInstance*, is accessible and deletable by anyone with the coordinates of the S3 server.
+
+<!TODO: check minio allows '/' in naming>
+- Flat Bucket Hierarchy:  Gluster-Swift's S3 implementation allows for nested buckets.
```

Comment:
I've checked Swift and any hierarchy is strictly visual by naming convention only. It, like all object stores (as far as I know), is flat.

Reply:
Thanks, I'll remove the bullet point.